### DISCUSSION TITLE
ListRepository page limit is hardcoded

### DISCUSSION THREAD

**Discussion by kurazsiosif** [quay/quay#3338] [Ideas]
We have noticed that the listRepositories can not be configured to limit the number of itmes per page:
https://github.com/quay/quay/blame/409d464ccab84d622c97be06847f78cddf1be256/endpoints/api/repository_models_pre_oci.py#L115

apparently this is hardcoded here, instead of providing the possibility in the API for the user to specify the limit quary paramter.
i think the correct implementation should have used the hardcoded value only if there is no limit quary parameter provided.

**Comment by GlitchPredator** [comment]
+1.

I noticed this problem too.

**Comment by razvan-serghescu-personal** [comment]
+1

**Comment by ibazulic** [comment]
The number of repositories returned is limited due to it being a heavy operation for the database. Methods here are used with the `GET /api/v1/repository` and that API endpoint returns a pagination token which can be used to navigate between pages. According to the API, the number of returned repos is 100, plus 1 extra entry for the pagination token itself.

**Reply by kurazsiosif** [reply]
We know this, but we want the possibility to limit it even further, like getting 1 repository per page. This way, we can ensure that the user-supplied authentication token has access to the repositories. However, our UI shows tag information for each repository, and we can't fit many repositories on one page. So, we need a finer split because we don't have an internal repository list buffer, which means we end up issuing a request each time the user changes the page—this is not ideal. There is no problem limiting the maximum number of items that can fit on one page.
